enhancements/machine-api/cluster-autoscaler-integration.md

```diff
@@ -0,0 +1,449 @@
+---
+title: cluster-autoscaler-integration
+authors:
+  - "@JoelSpeed"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-06-01
+last-updated: 2020-06-01
+status: implemented
+see-also:
+  - "/enhancements/machine-api/cluster-autoscaler-operator.md"  
+replaces:
+superseded-by:
+---
+
+# Cluster Autoscaler Integration
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The [Cluster Autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler) is a tool that users
+can deploy to a cluster to enable the automatic provisioning of extra resources when workloads exceed the cluster's
+capacity.
+It can also be configured to scale down resources if they are surplus to current requirement.
+
+In normal operation, a user would configure the Cluster Autoscaler to talk directly to their cloud provider and create
+new instances by expanding a nodegroup, such as an Autoscaling Group on AWS EC2.
+Because OpenShift users the Machine-API to create compute resources, OpenShift clusters do not have nodegroups in the
+way that the Cluster Autoscaler expects.
+
+This proposal details how a new provider will be implemented within the Cluster Autoscaler to integrate with Machine API
```

Comment:
The document blurs Machine API and Cluster API.  It would be good to draw a clearer distinction.

Across the broader Kubernetes ecosystem, it has been identified that having an infrastructure agnostic cluster autoscaler node group provider is a recurring challenge.  This is evidenced by the OpenShift Machine API, SAP Gardener, and Kubernetes-SIGS Cluster API efforts.  The only mechanism today for using the existing Cluster Autoscaler requires clients to either use the built-in infrastructure providers, or fork/extend the mainline repo (as many cloud providers do for their commercial integrations) to introduce their own extension.  This proposal tracks the work that was done to reduce the cost of fork/extend style extension in a downstream implementation.

A follow-on goal would be something like:
- mitigate the need to fork/extend by supporting a call-out mechanism and taking that upstream as well.